### PR TITLE
gitserver: expose clone and fetch errors in the UI

### PR DESCRIPTION
**Pull request by ryanslade**
This change ensures that we store clone and fetch errors in the database and exposes them to site
admins via the repo mirror settings area and also the repo list page.

Closes https://github.com/sourcegraph/sourcegraph/issues/32628

Screenshot 2022-06-15 at 12 21 40

Screenshot 2022-06-15 at 14 30 34

## Test plan

Unit tests added

### CODE REVIEW

**Comment by jasongornall** [comment]
Actually a couple of errors in buildkite you will want to take a look at
image

File: client/web/src/site-admin/SiteAdminRepositoriesPage.tsx [hunk 1]
```diff
@@ -67,6 +69,11 @@ const RepositoryNode: React.FunctionComponent<React.PropsWithChildren<Repository
                 }{' '}
             </div>
         </div>
+        {node.mirrorInfo.lastError && (
```
Comment (jasongornall) [inline client/web/src/site-admin/SiteAdminRepositoriesPage.tsx:72]: My only nitpick here is that we may surface different kinds of errors in the future so maybe choose a different variable name

**Comment by ryanslade** [comment]
@AlicjaSuska Thanks for the tips, see the new screenshots at the top.

**Comment by AlicjaSuska** [comment]
Thank you @ryanslade The first one looks good. In the second one I suggest to:
- add margin at the top (between the alert and settings button) - the same as bottom margin should be fine
- not sure if that's within the scope but radio buttons (all/cloned, etc.) look disorganised. Would it be possible to align rows to the left (so 'all' and 'needs index' are in the same line)?

**Comment by ryanslade** [comment]
> Thank you @ryanslade The first one looks good. In the second one I suggest to:
> 
> * add margin at the top (between the alert and settings button) - the same as bottom margin should be fine

Will do 👍 

> * not sure if that's within the scope but radio buttons (all/cloned, etc.) look disorganised. Would it be possible to align rows to the left (so 'all' and 'needs index' are in the same line)?

I'll see if that's possible

**Review by asdine — APPROVED** [review]

**Pull request merged**